snooty/test_project.py

```diff
@@ -417,6 +391,65 @@ def test_cache() -> None:
                     )
 
 
+def test_image_invalidation() -> None:
```

Comment:
awesome! TIL programmatically testing the cache. previous cache is not used for changed image 👍